libkbfs/interfaces.go

```diff
@@ -115,6 +115,22 @@ type Block interface {
 	ToCommonBlock() *CommonBlock
 	// IsIndirect indicates whether this block contains indirect pointers.
 	IsIndirect() bool
+	// OffsetExceedsData returns true if `off` is greater than the
+	// data contained in a direct block, assuming it starts at
+	// `startOff`.  Note that the offset of the next block isn't
+	// relevant; this function should only indicate whether the offset
+	// is greater than what currently could be stored in this block.
+	OffsetExceedsData(startOff, off Offset) bool
+	// BytesCanBeDirtied returns the number of bytes that should be
+	// marked as dirtied if this block is dirtied.
+	BytesCanBeDirtied() int64
+}
+
+// BlockWithPtrs defines methods needed for interacting with indirect
+// pointers.
+type BlockWithPtrs interface {
```

Comment:
Wouldn't it be better for this to be called `IndirectBlock` or `BlockIndirect`?

Reply:
It's doesn't necessarily represent an indirect block though.  It represents a block type that _can_ have indirect pointers (as opposed to `CommonBlock` and other test blocks).  But it can just as easily be a direct `DirBlock` or `FileBlock`, where `NumIndirectPtrs() == 0`.  I'm definitely open to a different name, but I don't think these names are right.